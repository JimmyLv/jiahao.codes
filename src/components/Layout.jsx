import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Header from './Header';

const LayoutDiv = styled.div`
  padding-left: 1rem;
  padding-right: 1rem;
`;

const ContentDiv = styled.div`
  margin: 3rem auto;
  max-width: 600px;
`;

export default function Layout({ path, children }) {
  return (
    <LayoutDiv>
      <Header path={path} />
      <ContentDiv>{children()}</ContentDiv>
    </LayoutDiv>
  );
}

Layout.propTypes = {
  path: PropTypes.string.isRequired,
  children: PropTypes.func.isRequired,
};
