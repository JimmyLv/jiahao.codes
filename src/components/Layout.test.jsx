import React from 'react';
import { shallow } from 'enzyme';
import Layout from './Layout';

test('Link changes the class when hovered', () => {
  const wrapper = shallow(
    <Layout path="/something">{() => <span>Some content</span>}</Layout>,
  );
  expect(wrapper).toMatchSnapshot();
});
