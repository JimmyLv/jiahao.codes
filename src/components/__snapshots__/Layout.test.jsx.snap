// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Layout
    path="/something"
  >
    [Function]
  </Layout>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Header
          path="/something"
        />,
        <styled.div>
          <span>
            Some content
          </span>
        </styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "path": "/something",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <span>
            Some content
          </span>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Some content",
          },
          "ref": null,
          "rendered": "Some content",
          "type": "span",
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Header
            path="/something"
          />,
          <styled.div>
            <span>
              Some content
            </span>
          </styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "path": "/something",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <span>
              Some content
            </span>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Some content",
            },
            "ref": null,
            "rendered": "Some content",
            "type": "span",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
